intake_netflow/v9.py

```diff
@@ -0,0 +1,521 @@
+    REPLICATION_FACTOR = 99
+    LAYER2_PACKET_SECTION_OFFSET = 102
+    LAYER2_PACKET_SECTION_SIZE = 103
+    LAYER2_PACKET_SECTION_DATA = 104
+
+
+class FieldDataType(enum.Enum):
+    IN_BYTES = int
+    IN_PKTS = int
+    FLOWS = int
+    PROTOCOL = int
+    SRC_TOS = int
+    TCP_FLAGS = int
+    L4_SRC_PORT = int
+    IPV4_SRC_ADDR = int
+    SRC_MASK = int
+    INPUT_SNMP = int
+    L4_DST_PORT = int
+    IPV4_DST_ADDR = int
+    DST_MASK = int
+    OUTPUT_SNMP = int
+    IPV4_NEXT_HOP = int
+    SRC_AS = int
+    DST_AS = int
+    BGP_IPV4_NEXT_HOP = int
+    MUL_DST_PKTS = int
+    MUL_DST_BYTES = int
+    LAST_SWITCHED = int
+    FIRST_SWITCHED = int
+    OUT_BYTES = int
+    OUT_PKTS = int
+    MIN_PKT_LNGTH = int
+    MAX_PKT_LNGTH = int
+    IPV6_SRC_ADDR = bytes
+    IPV6_DST_ADDR = bytes
+    IPV6_SRC_MASK = int
+    IPV6_DST_MASK = int
+    IPV6_FLOW_LABEL = bytes
+    ICMP_TYPE = int
+    MUL_IGMP_TYPE = int
+    SAMPLING_INTERVAL = int
+    SAMPLING_ALGORITHM = int
+    FLOW_ACTIVE_TIMEOUT = int
+    FLOW_INACTIVE_TIMEOUT = int
+    ENGINE_TYPE = int
+    ENGINE_ID = int
+    TOTAL_BYTES_EXP = int
+    TOTAL_PKTS_EXP = int
+    TOTAL_FLOWS_EXP = int
+    IPV4_SRC_PREFIX = int
+    IPV4_DST_PREFIX = int
+    MPLS_TOP_LABEL_TYPE = int
+    MPLS_TOP_LABEL_IP_ADDR = int
+    FLOW_SAMPLER_ID = int
+    FLOW_SAMPLER_MODE = int
+    FLOW_SAMPLER_RANDOM_INTERVAL = int
+    MIN_TTL = int
+    MAX_TTL = int
+    IPV4_IDENT = int
+    DST_TOS = int
+    IN_SRC_MAC = bytes
+    OUT_DST_MAC = bytes
+    SRC_VLAN = int
+    DST_VLAN = int
+    IP_PROTOCOL_VERSION = int
+    DIRECTION = int
+    IPV6_NEXT_HOP = bytes
+    BPG_IPV6_NEXT_HOP = bytes
+    IPV6_OPTION_HEADERS = int
+    MPLS_LABEL_1 = bytes
+    MPLS_LABEL_2 = bytes
+    MPLS_LABEL_3 = bytes
+    MPLS_LABEL_4 = bytes
+    MPLS_LABEL_5 = bytes
+    MPLS_LABEL_6 = bytes
+    MPLS_LABEL_7 = bytes
+    MPLS_LABEL_8 = bytes
+    MPLS_LABEL_9 = bytes
+    MPLS_LABEL_10 = bytes
+    IN_DST_MAC = bytes
+    OUT_SRC_MAC = bytes
+    IF_NAME = str
+    IF_DESC = str
+    SAMPLER_NAME = str
+    IN_PERMANENT_BYTES = int
+    IN_PERMANENT_PKTS = int
+    FRAGMENT_OFFSET = int
+    FORWARDING_STATUS = int
+    MPLS_PAL_RD = bytes
+    MPLS_PREFIX_LEN = int
+    SRC_TRAFFIC_INDEX = int
+    DST_TRAFFIC_INDEX = int
+    APPLICATION_DESCRIPTION = str
+    APPLICATION_TAG = bytes
+    APPLICATION_NAME = str
+    POST_IP_DIFF_SERV_CODE_POINT = int
+    REPLICATION_FACTOR = int
+    LAYER2_PACKET_SECTION_OFFSET = int
+    LAYER2_PACKET_SECTION_SIZE = int
+    LAYER2_PACKET_SECTION_DATA = bytes
+
+
+@attr.s
+class Header(object):
+    version = attr.ib(type=int, default=9)
+    count = attr.ib(type=int, default=0)
+    uptime = attr.ib(type=int, default=0)
+    datetime = attr.ib(type=int)
+    sequence = attr.ib(type=int, default=0)
+    source_id = attr.ib(type=int, default=0)
+
+    @datetime.default
+    def current_unix_seconds(self):
+        return int(datetime.utcnow().strftime("%s"))
```

Comment:
The Netflow spec mandates that each packet has the number of seconds since UTC 0, so `time.time()` will work. I made the timestamp optional in the packet header constructor for ease of use (definitely helps for testing).